### DISCUSSION TITLE
Aperture weight to exposure time conversion

### DISCUSSION THREAD

**Discussion by ehssal** [e0404/matRad#523] [Q&A]
Hi,

I am looking for a way to calculate the corresponding exposure times for aperture weights/intensities optimized in matRad (for photons). Is the beamlet/aperture weight unit already stored in matRad data structures or is there a built-in function to calculate it?

**Comment by wahln** [comment]
Hi @ehssal,
if you follow the matRad.m script, you see that results of fluence optimizations are stored in the resultGUI. First, this will contain weights/intensities per beamlet/bixel, in the positive real domain, and not applicable shapes.
Afterwards, you can perform sequencing choosing one of the available sequencing algorithms to create apertures with weights. The matRad.m script uses the siochi sequencer here with 5 stratification levels: https://github.com/e0404/matRad/blob/3d29b765a9a0a02a593890726102a7d9db907af7/matRad.m#L73
These apertures are then stored  resultGUI.sequencing _and_ resultGUI.apertureInfo (the latter contains a little more information for direct aperture optimization / possible VMAT planning). There you find shape structure in matrix form, weights, and so on)

From the GUI you can perform sequencing by activating "Run Sequencing" before optimization and setting the number of stratification levels.

**Reply by ehssal** [reply]
Hi Niklas,

Thank you for your response. My question is related to after the sequencing step. Specifically, once the sequencing is done in matRad, how can one convert the weights of the apertures into time (i.e., seconds)?

**Reply by eric11210** [reply]
The way I did it for the VMAT implementation was to first convert aperture weight to MU, then MU to time.

For conversion to aperture weight to MU, you need to know the calibration condition of your beam (eg 10 cm SAD, 10x10 field), and the dose/MU at that reference point. Then calculate what weight in matRad is needed to achieve that dose in the same reference field.

The conversion from MU to time is simply done through the MU rate of your machine (eg 600 MU/s).

Hope that helps!
